src/java/htsjdk/samtools/reference/IndexedFastaSequenceFile.java

```diff
@@ -202,10 +251,20 @@ public ReferenceSequence getSubsequenceAt( String contig, long start, long stop
             startOffset += Math.max((int)(startOffset%bytesPerLine - basesPerLine + 1),0);
 
             try {
-                 startOffset += channel.read(channelBuffer,indexEntry.getLocation()+startOffset);
+                if (channel instanceof FileChannel) {
+                    startOffset += ((FileChannel) channel).read(channelBuffer,indexEntry.getLocation()+startOffset);
+                } else {
+                    long oldPos = channel.position();
+                    try {
+                        channel.position(indexEntry.getLocation()+startOffset);
+                        startOffset += channel.read(channelBuffer);
+                    } finally {
+                        channel.position(oldPos);
+                    }
+                }
             }
             catch(IOException ex) {
-                throw new SAMException("Unable to load " + contig + "(" + start + ", " + stop + ") from " + file);
+                throw new SAMException("Unable to load " + contig + "(" + start + ", " + stop + ") from " + getAbsolutePath());
```

Comment:
Could you add `ex` as the cause to this exception.  I'm worried there is a case where the finally clause will throw and without the original exception it won't be clear what's going on.